app/controllers/users_controller.rb

```diff
@@ -0,0 +1,32 @@
+class UsersController < ApplicationController
+  # before_action :user_agreed?, only: [:create]
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    session[:user_id] = @user.id
+    if @user.save
+      redirect_to home_index_path
+    else
+      flash[:danger] = @user.errors.full_messages.first
```

Comment:
Might be cool to send over all of the flash messages instead of just the first and then in the view iterate over them using `content_tag`. Something like this:

```ruby
flash.each do |msg_type, message|
  content_tag(:div, message, class: msg_type)
end
```